driftbase/api/parties.py

```diff
@@ -0,0 +1,352 @@
+        }
+        return response, http_client.OK
+
+    def delete(self, invite_id):
+        player_id = current_user['player_id']
+        inviter_id, invited_id = decline_party_invite(invite_id, player_id)
+        if inviter_id == player_id:
+            log.debug("Player {} canceled a party invite to {}".format(inviter_id, invited_id))
+            _add_message("players", invited_id, "party_notification",
+                         {
+                             "event": "invite_canceled",
+                             "invite_id": invite_id,
+                             "inviting_player_id": inviter_id,
+                             "inviting_player_url": url_for("players.entry", player_id=inviter_id, _external=True),
+                         })
+        else:
+            log.debug("Player {} declined a party invite from {}".format(player_id, inviter_id))
+            _add_message("players", inviter_id, "party_notification",
+                         {
+                             "event": "invite_declined",
+                             "player_id": invited_id,
+                             "player_url": url_for("players.entry", player_id=invited_id, _external=True),
+                         })
+        return {}, http_client.NO_CONTENT
+
+
+@bp_parties.route("/", endpoint="list")
+class PartiesAPI(MethodView):
+    """
+    Return the party for the current player
+    """
+
+    @bp_parties.arguments(PartyGetSchema)
+    def get(self, args):
+        player_id = current_user['player_id']
+        party_id = get_player_party(player_id)
+
+        if party_id is None:
+            abort(http_client.NOT_FOUND)
+
+        party_id = int(party_id)
+        party_members = get_party_members(party_id)
+        member_query = g.db.query(CorePlayer.player_id, CorePlayer.player_name).filter(CorePlayer.player_id.in_(party_members))
+        response, response_header = make_party_response(party_id, member_query.all())
+        return response, http_client.OK, response_header
+
+
+@bp_parties.route("/<int:party_id>/", endpoint="entry")
+class PartyAPI(MethodView):
+    """
+    Manage party of players.
+    """
+
+    def get(self, party_id):
+        members = get_party_members(party_id)
+        if not members:
+            abort(http_client.NOT_FOUND)
+
+        player_id = current_user['player_id']
+        if player_id not in members:
+            abort(http_client.FORBIDDEN, message="This is not your party")
+
+        member_query = g.db.query(CorePlayer.player_id, CorePlayer.player_name).filter(CorePlayer.player_id.in_(members))
+        response, response_header = make_party_response(party_id, member_query.all())
+        return response, http_client.OK, response_header
+
+    def delete(self, party_id):
+        members = get_party_members(party_id)
+        for member in members:
+            leave_party(party_id, member)
+        return {}, http_client.NO_CONTENT
```

Comment:
Do we not want to delete the party after removing the last player from it?

Reply:
Yes that would make sense, fixing...